raft/src/main/java/org/apache/kafka/raft/LeaderState.java

```diff
@@ -117,9 +117,10 @@ public long timeUntilCheckQuorumExpires(long currentTimeMs) {
         long remainingMs = checkQuorumTimer.remainingMs();
         if (remainingMs == 0) {
             log.info(
-                "Did not receive fetch request from the majority of the voters within {}ms. Current fetched voters are {}.",
+                "Did not receive fetch request from the majority of the voters within {}ms. Current fetched voters are {}, and voters are {}",
                 checkQuorumTimeoutMs,
-                fetchedVoters);
+                fetchedVoters,
+                voterStates.keySet());
```

Comment:
Let's print the replica keys (id and directory id) not just the ids. E.g. `votersStates.values().stream.map(ReplicaState::replicaKey)`.

Having said, my comment doesn't apply here since trunk doesn't have those changese. I am thinking that maybe we should wait for us to merge https://github.com/apache/kafka/pull/16235.

Reply:
OK, will review https://github.com/apache/kafka/pull/16235 later.